library/src/main/java/com/pokegoapi/util/PokeNames.java

```diff
@@ -41,4 +42,53 @@ public static String getDisplayName(int pokedexNr, Locale locale) {
 			}
 		return str;
 	}
+
+	/**
+	 * Returns translated Pokemon name from english locale.
+	 * @param engName pokemon english name
+	 * @param newLocale target locale
+	 * @return translated pokemon name
+	 */
+	public static String translateName(String engName, Locale newLocale) {
+		ResourceBundle engNameList = ResourceBundle.getBundle("pokemon_names");
+		ResourceBundle translatedNameList = ResourceBundle.getBundle("pokemon_names", newLocale);
+
+		String nameKey = "", translatedName = "", engNameUTF8 = "";
```

Comment:
To reduce the duplicate logic of this method you could evaluate if its possible for you to split it into two.